### DISCUSSION TITLE
How to enforce size to the image icon?

### DISCUSSION THREAD

**Discussion by dikesh** [elkowar/eww#1233] [Q&A]
I am trying to set icon using `image` widget but the icon size is too large.
How can I resize / scale the same to fit the parent container / widget?

```
; Power widget
(defwidget power []
  (button
    :class "bar-section power"
    (image :icon "system-shutdown-symbolic")))

; Bar right
(defwidget bar-right [] 
  (box
    :space-evenly false
    :spacing 8
    :halign "end"
    (image 
      :image-height 16
      :image-width 16
      :preserve-aspect-ratio true
      :icon-size "small-toolbar"
      :icon "localsend")
    (power)))
```
### Output
![image](https://github.com/user-attachments/assets/d433de45-05f2-4fe3-b339-48ea98cc1e4b)

**Comment by Chlebqowy** [comment]
Is that still the case after making the values smaller?

**Reply by dikesh** [reply]
Yes. I tried all the properties with different values but nothing is working.
I am using Papirus icon theme but it doesn't have icon for `localsend` yet. So eww is picking icon from `hicolor` where smallest available size is 128x123

### fzf output

![image](https://github.com/user-attachments/assets/55798fdf-a26f-4a59-932a-2a4bcffa9adb)

**Comment by 14hi0n** [comment]
Have you found a solution? I have the same problem. Image ignores dimensions

**Reply by dikesh** [reply]
Nope... I wanted to switch from `ags` to `eww` but I felt animation was bit laggy compared to `ags`. 
So  I ended up switching to [ignis](https://ignis-sh.github.io/ignis/stable/index.html).

**Comment by LurkAndLoiter** [comment]
Feature Request [1292](https://github.com/elkowar/eww/issues/1292) The solution is to just provide the hard link path to the icon  and treat it as an image and then use image-height image-width. This however is not viable workaround for dynamic icon usage. I'm not sure if it's intentional or not but image-height and image-width do nothing for icons only icon-size has any affect and it has issues if you have any missing sizes for example if the icon image is only available in 256x256 then that's what you're stuck with it will not resize that image if you're treating it as an :icon and not a :path.

```lisp
; something like this for images
:path "/here.png"
:image-height 16
:image-width 16

; something like this for icons
:icon "some-icon"
:icon-size "dialog"
```
